Tests/BSONTests/DocumentTests.swift

```diff
@@ -345,249 +296,121 @@ final class DocumentTests: MongoSwiftTestCase {
-            "int32": "hi",
-            "int64": .decimal128(Decimal128("1.0")!),
-            "bool": [1, 2, 3],
-            "decimal": 100,
-            "oid": 25.5,
-            "timestamp": .objectId(newOid),
-            "datetime": .timestamp(Timestamp(timestamp: 1, inc: 2))
+        expect(doc).to(equal([
+            "string": false,
+            "true": "hi",
+            "false": 1000,
+            "int": true,
+            "int32": .int64(4),
+            "int64": .int32(15),
+            "double": .decimal128(try BSONDecimal128("100")),
+            "decimal128": 3.0,
+            "minkey": .maxKey,
+            "maxkey": .minKey,
+            "date": .timestamp(BSONTimestamp(timestamp: 20, inc: 30)),
+            "timestamp": .datetime(Date(msSinceEpoch: 2000)),
+            "nestedarray": ["e": 5, "f": 6],
+            "nesteddoc": [[3, 4], [.int32(5), .int32(6)]],
+            "oid": .regex(BSONRegularExpression(pattern: "xyz$", options: "ix")),
+            "regex": .objectID(newOid),
+            "array1": [10, 11, 12],
+            "array2": .null,
+            "null": ["zzzzzz"],
+            "code": .codeWithScope(BSONCodeWithScope(code: "console.log(z);", scope: ["z": 100])),
+            "codewscope": .code(BSONCode(code: "console.log('bye');"))
         ]))
-
-        // make a deep copy so we start off with uniquely referenced storage
-        var nonOverwritableDoc = DocumentTests.nonOverwritables.withBSONPointer { ptr in
-            Document(copying: ptr)
-        }
-
-        // save a reference to original bson_t so we can verify it changes
-        var nonOverwritablePointer = nonOverwritableDoc.pointerAddress
-
-        let nonOverwritablePairs: [(String, BSON)] = [("string", 1), ("nil", "hello"), ("doc", "hi"), ("arr", 5)]
-
-        nonOverwritablePairs.forEach { k, v in
-            nonOverwritableDoc[k] = v
-            expect(nonOverwritableDoc.pointerAddress).toNot(equal(nonOverwritablePointer))
-            nonOverwritablePointer = nonOverwritableDoc.pointerAddress
-        }
-
-        expect(nonOverwritableDoc).to(equal(["string": 1, "nil": "hello", "doc": "hi", "arr": 5]))
     }
 
     // test setting both overwritable and nonoverwritable values to nil
-    func testReplaceValueWithNil() throws {
-        var overwritableDoc = DocumentTests.overwritables.withBSONPointer { ptr in
-            Document(copying: ptr)
-        }
-        var overwritablePointer = overwritableDoc.pointerAddress
-
-        ["double", "int32", "int64", "bool", "decimal", "oid", "timestamp", "datetime"].forEach {
-            overwritableDoc[$0] = .null
-            // the storage should change every time
-            expect(overwritableDoc.pointerAddress).toNot(equal(overwritablePointer))
-            overwritablePointer = overwritableDoc.pointerAddress
-        }
-
-        var nonOverwritableDoc = DocumentTests.nonOverwritables.withBSONPointer { ptr in
-            Document(copying: ptr)
+    func testDeletions() throws {
```

Comment:
this was replacing everything with `null`, but I think the intent of the test was really probably to test deletions, and it got garbled when we did a project to make it so users could differentiate between null and nil. so I've switched it to test deletions instead